src/core/lib/iomgr/tcp_server_posix.cc

```diff
@@ -223,6 +220,12 @@ static void on_read(void* arg, grpc_error_handle err) {
       }
     }
 
+    if (sp->server->memory_quota->IsMemoryPressureHigh()) {
```

Comment:
Ah! I see the backpressure logic was moved. Is there any harm in moving it from the transport to the i/o layer? Seems like it rejects connections later than it did before, allowing a few more allocations to occur maybe, but I'm not sure that's much of a concern.

Reply:
I don't think there's any harm in doing this, and it does allow us to reduce work in bad situations, so I'd like to try it.